src/main/java/org/marketdesignresearch/mechlib/utils/CPLEXUtils.java

```diff
@@ -20,10 +19,19 @@
     private final Map<SolveParam, Object> solveParamMap = Maps.newHashMap();
 
     public IMIPResult solve(IMIP program) {
+        return solve(program, Map.of());
+    }
+
+    public IMIPResult solve(IMIP program, Map<SolveParam, Object> additionalParams) {
         solveParamMap.forEach(program::setSolveParam);
+        additionalParams.forEach(program::setSolveParam);
         return solver.solve(program);
     }
 
+    public void setSolveParam(SolveParam param, Object value) {
+        solveParamMap.put(param, value);
+    }
+
     public void initializeSolveParams() {
         solveParamMap.put(SolveParam.CALC_DUALS, Boolean.TRUE);
         solveParamMap.put(SolveParam.LP_OPTIMIZATION_ALG, 2);
```

Comment:
There are a lot of SolveParams set which are might too specific for a global setting. These were orginally only used to calculate the Norm MIPs and may should be moved to this location.

Reply:
I agree!